src/lib/menu/menu-directive.ts

```diff
@@ -238,19 +238,27 @@ export class MatMenu implements AfterContentInit, MatMenuPanel<MatMenuItem>, OnI
   }
 
   ngAfterContentInit() {
-    this._keyManager = new FocusKeyManager<MatMenuItem>(this._items).withWrap().withTypeAhead();
+    this._allItems.changes
```

Comment:
Add comment here that explains what you're doing (collecting direct descendants) and why they're needed? Might also be good to break this out into its own method like `_updateDirectDescendants`